src/replay/execution.zig

```diff
@@ -194,8 +200,14 @@ fn replaySlot(state: ReplayExecutionState, slot: Slot) !ReplaySlotStatus {
         return .dead;
     }
 
-    const epoch_info = state.epochs.getForSlot(slot) orelse return error.MissingEpoch;
-    const slot_info = state.slot_tracker.get(slot) orelse return error.MissingSlot;
+    const epoch_info_ptr: *const sig.core.EpochConstants = blk: {
+        const ep, var ep_lg = state.epochs.readWithLock();
+        defer ep_lg.unlock();
+        break :blk ep.getPtrForSlot(slot) orelse return error.MissingEpoch;
+    };
```

Comment:
The pointer lifetime exceeds the duration of the lock which violates the intended protection of the lock.

Reply:
Updated to use the lock to create a clone of the epoch constants [618d05f](https://github.com/Syndica/sig/pull/875/commits/618d05f4f3c5091b4bc4dbfdff485d3dd8a82ff2)

Given the epoch constants should not change for a given slot/epoch I think this is semantically correct. Although I am not sure if there might be a better way to do this (one that avoids a clone for example)

Reply:
I would have just held the lock for more time, but this works too.

Reply:
8c66e61